File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.cpp
class Solution {
public:
    int closestTarget(vector<string>& words, string target, int start) {
        int n = words.size();
        int ans = INT_MAX;
        for(int i = 0; i < n; i++){
            if(words[i] ==target){
                int distance= abs(i - start);
                ans = min(ans, min(distance, n - distance));
            }
        }
       if (ans == INT_MAX) {
        return -1;
            } else {
             return ans;
                }
    }
}
;

File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.java
class Solution {
    public int closestTarget(String[] words, String target, int startIndex) {
        int n = words.length;
        int ans = Integer.MAX_VALUE;
        for (int i = 0; i < n; i++) {
            if (words[i].equals(target)) {
                int distance = Math.abs(i - startIndex);
                ans = Math.min(ans, Math.min(distance, n - distance));
            }
        }
        if (ans == Integer.MAX_VALUE) {
            return -1;
        } else {
            return ans;
        }
    }
}